compiler/rustc_hir/src/hir.rs

```diff
@@ -1371,7 +1373,7 @@ pub struct Expr<'hir> {
 
 // `Expr` is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(target_arch = "x86_64")]
-rustc_data_structures::static_assert_size!(Expr<'static>, 72);
+rustc_data_structures::static_assert_size!(Expr<'static>, 144);
```

Comment:
Would replacing `AnonConst` by `&'hir AnonConst<'hir>` help with the size?

Reply:
yeah, probably :+1: 

I want to wait for a first review before looking into perf though, so I don't waste time on an approach we don't want in the end